src/main/scala/repos/jdbc/JdbcDb.scala

```diff
@@ -35,27 +35,36 @@ class JdbcDb(val profile: JdbcProfile, private[repos] val db: JdbcProfile#Backen
       i.name -> new IndexTableImpl(i)(i.projectionType)
     }.toMap
 
-    def insert(elements: Iterable[(Id, M)], skipIndexesForTesting: Boolean = false)(implicit ec: ExecutionContext): Future[Seq[Long]] = {
+    def insert(elements: Iterable[(Id, M)], skipIndexesForTesting: Boolean = false, insertLatest: Boolean = true)(implicit ec: ExecutionContext): Future[Seq[Long]] = {
       val newEntries = elements.map(e => EntryTableRecord[Id, M](-1, e._1, System.currentTimeMillis(), e._2))
 
       val effectiveIndexMap =
         if (skipIndexesForTesting) Map.empty[String, IndexTableImpl[_]] else
         indexMap
       // Build the insert as a single DBAction.
-      val r = for {
-        pkList <- entryTable.returning(entryTable.map(_.pk)) ++= newEntries
-        _ <- DBIO.sequence(effectiveIndexMap.values.map(
-          _.buildInsertAction(elements zip pkList)))
-        // As we are making all the inserts in one big transaction which locks the rows being updated
-        // We must make sure that we make the inserts in the same order. Otherwise, a parallel insertion
-        // that tries to insert the same ids but in a different order will cause a deadlock.
-        // To avoid this situation, we insert the new elements into the latest table ordered by
-        // their ids.
-        _ <- DBIO.sequence((newEntries zip pkList).toSeq.sortBy(t => idMapper.toUUID(t._1.id)).map {
-          e => latestEntryTable.insertOrUpdate((e._1.id, e._1.entry, e._2))
-        })
-      } yield pkList
-      db.run(r.transactionally)
+      if (insertLatest) {
+        val r = for {
+          pkList <- entryTable.returning(entryTable.map(_.pk)) ++= newEntries
```

Comment:
Can it be factored out in a separate function such that in case of insertLatest it is just composed with DBIO.sequence(...) block?